.github/workflows/linux-validation.yml

```diff
@@ -0,0 +1,99 @@
+name: Linux Validation
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  linux-validation:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    
+    - name: Install system dependencies
+      run: |
+        sudo apt update
+        sudo apt install -y git git-lfs build-essential cmake gcc make curl
+        # Set up Git LFS
+        git lfs install
+    
+    - name: Install PowerShell 7
+      run: |
+        # Download and install PowerShell 7
+        wget -q "https://packages.microsoft.com/config/ubuntu/$(lsb_release -rs)/packages-microsoft-prod.deb"
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt update
+        sudo apt install -y powershell
+        # Verify PowerShell installation
+        pwsh -c "Write-Host 'PowerShell installed successfully'"
+    
+    - name: Install Rust toolchain
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: 1.87
+        components: rustfmt, clippy
+    
+    - name: Install nightly Rust toolchain
+      uses: dtolnay/rust-toolchain@nightly
+      with:
+        components: miri, rustfmt
+    
+    - name: Setup Rust cache
+      uses: Swatinem/rust-cache@v2
+      with:
+        cache-on-failure: true
+        shared-key: "folo-linux"
+    
+    - name: Cache cargo tools
+      uses: actions/cache@v4
+      with:
+        path: ~/.cargo/bin
+        key: cargo-tools-linux-${{ hashFiles('just_setup.just') }}
+        restore-keys: |
+          cargo-tools-linux-
+    
+    - name: Install just
+      run: cargo install just --locked
+    
+    - name: Install development tools
+      run: |
+        # Install tools with retry mechanism and timeout handling
+        timeout 1800 just install-tools || {
+          echo "Installation timed out or failed, retrying with individual components..."
```

Comment:
There is no need to have two ways to install things. `just install-tools` must always succeed, there is no value in having a fallback path here.